angular-app/src/app/pages/manage-room/manage-room.component.html

```diff
@@ -2,54 +2,65 @@
 <div class="container">
   <button mat-raised-button color="primary" class="back-btn"> ＜ 戻る</button>
   <div class="edit-form">
-    <mat-form-field class="name" appearance="fill">
+    <mat-form-field class="name" hintLabel="Max 20 characters" appearance="fill">
       <mat-label>部屋名</mat-label>
-      <input matInput>
+      <input matInput #input maxlength="20" [formControl]="roomName">
+      <mat-hint align="end">{{input.value.length || 0}}/20</mat-hint>
     </mat-form-field>
-    <button mat-raised-button color="primary" class="save-room-btn">保存</button>
+    <button mat-raised-button color="primary" class="save-room-btn" [disabled]="roomName.invalid" (click)="saveRoom()">{{saveRoomBtnName}}</button>
```

Comment:
ボタンのtextは`roomId`で判断してそのまま表示させちゃいましょう
```suggestion
    <button mat-raised-button color="primary" class="save-room-btn" [disabled]="roomName.invalid" (click)="saveRoom()">{{ roomId ? '登録': '更新' }}</button>
```
お題の方も同様ですね